File: readability/readability.c
#include <cs50.h>
#include <stdio.h>
#include <math.h>
#include <string.h>
#include <ctype.h>

int count_letters(string s);

int count_words(string s);

int count_sentences(string s);
int main(void)
{
    string text = get_string("Text: ");
    float L = (count_letters(text) * 100.00) / count_words(text); // Number of letters in 100 words
    float S = (count_sentences(text) * 100.00) / count_words(text); // Number of sentences in 100 words
    float index = (0.0588 * L) - (0.296 * S) - 15.8 ;// Coleman-Liau index.
    if (index < 1)
    {
        printf("Before Grade 1\n");
    }
    else if (index > 16)
    {
        printf("Grade 16+\n");
    }
    else
    {
        index = round(index); // Rounding off to nearest integer
        printf("Grade %i\n", (int) index);
    }
}


int count_letters(string s) // Function to return letters in a text
{
    int total_letters = 0;
    for (int i = 0; i < strlen(s); i++)
    {
        if (isalpha(s[i]))
        {
            total_letters++;
        }
    }
    return total_letters;
}

int count_words(string s) // Function to return words in a text
{
    int total_words = 1;
    for (int i = 0; i < strlen(s); i++)
    {
        if (isspace(s[i]) && ! isspace(s[i - 1])  && ! isspace(s[i + 1]))
        {
            total_words++;
        }
    }
    return total_words;
}

int count_sentences(string s) // Funtion to return se ntences in a text
{
    int total_sentences = 0;
    for (int i = 0; i < strlen(s); i++)
    {
        if (s[i] == '.' || s[i] == '!' || s[i] == '?')
        {
            total_sentences++;
        }
    }
    return total_sentences;
}

File: sentimental-readability/readability.py
import re
text = input("Text: ")
words = text.split()  # spliting text into words.
lines = re.split(r"[.!?]", text)  # spliting text into part at (.,!,?).
letters_count = 0
for i in text:  # Counting letters in text.
    if i.isalpha():
        letters_count += 1
L = float(letters_count / len(words)) * 100  # Average number of letters per 100 words in the text.
S = float((len(lines) - 1) / len(words)) * 100  # Average number of sentences per 100 words in the text.
Grade = round(0.0588 * L - 0.296 * S - 15.8)  # Coleman-Liau index formula.
if Grade < 1:  # If Grade less than 1.
    print("Before Grade 1")
elif Grade > 16:  # If Grade is more than 16.
    print("Grade 16+")
else:  # If Grade is between 1 and 16 (including them).
    print(f" Grade {Grade}")